File: library/src/main/java/com/benny/library/neobinding/adapter/SimpleAdapterItemAccessor.java
package com.benny.library.neobinding.adapter;

import java.util.List;

/**
 * Created by benny on 9/18/15.
 */
public class SimpleAdapterItemAccessor<T> implements AdapterItemAccessor<T> {
    private List<T> list;

    public SimpleAdapterItemAccessor(List<T> list) {
        this.list = list;
    }

    public int size() {
        return list.size();
    }
    public T get(int position) {
        return list.get(position);
    }

    @Override
    public boolean isEmpty() {
        return list.isEmpty();
    }
}

File: library/src/main/kotlin/com/benny/library/neobinding/kotlin/adapter/SimpleAdapterItemAccessor.kt
package com.benny.library.neobinding.kotlin.adapter

/**
 * Created by benny on 9/18/15.
 */
class SimpleAdapterItemAccessor(private val list: List<*>) : AdapterItemAccessor {

    override fun size(): Int {
        return list.size
    }

    override fun get(position: Int): Any {
        return list[position] as Any
    }

    override fun isEmpty(): Boolean {
        return list.isEmpty()
    }
}
